pallets/scheduler/src/lib.rs

```diff
@@ -616,310 +710,57 @@ pub mod pallet {
-
-								Some(Scheduled {
-									maybe_id: schedule.maybe_id.map(|x| blake2_256(&x[..])),
-									priority: schedule.priority,
-									call,
-									maybe_periodic: schedule.maybe_periodic,
-									origin: schedule.origin,
-									_phantom: Default::default(),
-								})
-							})
-							.or_else(|| {
-								log::info!("Schedule in agenda for block {:?} is empty - nothing to do here.", &block);
-								None
-							})
-                    })
-                    .collect::<Vec<_>>(),
-            ))
-        });
-
-        #[allow(deprecated)]
-        frame_support::storage::migration::remove_storage_prefix(
-            Self::name().as_bytes(),
-            b"StorageVersion",
-            &[],
-        );
-
-        StorageVersion::new(4).put::<Self>();
-
-        weight + T::DbWeight::get().writes(2)
-    }
-}
-
-impl<T: Config> Pallet<T> {
-    /// Helper to migrate scheduler when the pallet origin type has changed.
-    pub fn migrate_origin<OldOrigin: Into<T::PalletsOrigin> + codec::Decode>() {
-        Agenda::<T>::translate::<
-            Vec<
-                Option<
-                    Scheduled<
-                        TaskName,
-                        BoundedCallOf<T>,
-                        BlockNumberFor<T>,
-                        OldOrigin,
-                        T::AccountId,
-                    >,
-                >,
-            >,
-            _,
-        >(|_, agenda| {
-            Some(BoundedVec::truncate_from(
-                agenda
-                    .into_iter()
-                    .map(|schedule| {
-                        schedule.map(|schedule| Scheduled {
-                            maybe_id: schedule.maybe_id,
-                            priority: schedule.priority,
-                            call: schedule.call,
-                            maybe_periodic: schedule.maybe_periodic,
-                            origin: schedule.origin.into(),
-                            _phantom: Default::default(),
-                        })
-                    })
-                    .collect::<Vec<_>>(),
-            ))
-        });
-    }
-
     fn resolve_time(
-        when: DispatchTime<BlockNumberFor<T>>,
-    ) -> Result<BlockNumberFor<T>, DispatchError> {
-        let now = frame_system::Pallet::<T>::block_number();
+        when: DispatchTime<BlockNumberFor<T>, T::Moment>,
+    ) -> Result<BlockNumberOrTimestampOf<T>, DispatchError> {
+        let current_block = frame_system::Pallet::<T>::block_number();
+        let now = T::TimeProvider::now();
 
         let when = match when {
-            DispatchTime::At(x) => x,
+            DispatchTime::At(x) => BlockNumberOrTimestamp::BlockNumber(x),
             // The current block has already completed it's scheduled tasks, so
             // Schedule the task at lest one block after this current block.
-            DispatchTime::After(x) => now.saturating_add(x).saturating_add(One::one()),
+            DispatchTime::After(x) => {
```

Comment:
Can you explain why you can't just schedule in next block here even if it was originally scheduled via time?